agent/proxycfg/snapshot.go

```diff
@@ -108,6 +106,52 @@ func (c *configSnapshotMeshGateway) IsEmpty() bool {
 		len(c.ConsulServers) == 0
 }
 
+type configSnapshotIngressGateway struct {
+	// Config is the associated ingress-gateway config entry that we are generating watches for.
+	Config *structs.IngressGatewayConfigEntry
+
+	// ServiceLists is a map of maps, the first key being a namespace and the
+	// second being the service ID. This is used to separate out all services by
+	// namespace in order to deal with "*" wildcard specifiers.
+	ServiceLists map[string]map[structs.ServiceID]struct{}
```

Comment:
Since `structs.ServiceID` itself contains a namespace, would it make more sense to just have this be a `map[string]map[string]struct{}` instead rather than duplicating the NS in each entry?

Reply:
We've removed this field from the ConfigSnapshot since we've decided to go with a different RPC mechanism that sends only the required set of services to watch.

Thus, we don't need to maintain a list of all services by namespace anymore.